sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ModelTests.cs

```diff
@@ -71,12 +72,82 @@ protected void RoundTripTest(string format, RoundTripStrategy<T> strategy)
             var data = strategy.Write(model, options);
             string roundTrip = data.ToString();
 
-            Assert.That(roundTrip, Is.EqualTo(expectedSerializedString));
+            // we validate those are equivalent element, representing the same json object (ignoring the spaces and orders, etc)
+            AssertJsonEquivalency(expectedSerializedString, roundTrip);
 
             T model2 = (T)strategy.Read(roundTrip, ModelInstance, options);
             CompareModels(model, model2, format);
         }
 
+        private void AssertJsonEquivalency(string expected, string result)
+        {
+            using var expectedDoc = JsonDocument.Parse(expected);
+            using var resultDoc = JsonDocument.Parse(result);
+
+            AssertJsonEquivalency(expectedDoc.RootElement, resultDoc.RootElement);
+        }
+
+        private void AssertJsonEquivalency(JsonElement expected, JsonElement result)
+        {
+            if (expected.ValueKind != result.ValueKind)
+            {
+                Assert.Fail($"kind does not match between {expected} and {result}");
+            }
+
+            switch (result.ValueKind)
+            {
+                case JsonValueKind.Object:
+                    AssertJsonObjectEquivalency(expected, result);
+                    break;
+                case JsonValueKind.Array:
+                    AssertJsonArrayEquivalency(expected, result);
+                    break;
+                default:
+                    Assert.AreEqual(expected.ToString(), result.ToString());
```

Comment:
Would this be correct for something like a bool or int? It seems that if the extected value is bool with value true, but the actual is the string "true", this would still return that they are equivalent.

Reply:
Nevermind, I see that there would be different ValueKinds in that case and the assertion would fail on L94.

Reply:
I tried if we could just assert they are equal without toString, but it does not work.
Maybe `GetRawText` is a better solution here?